src/socket/tcp_socket.cr

```diff
@@ -39,6 +68,15 @@ class TCPSocket < IPSocket
   # eventually closes the socket when the block returns.
   #
   # Returns the value of the block.
+  def self.open(host, port, local_host, local_port)
+    sock = new(host, port, local_host: local_host, local_port: local_port)
+    begin
+      yield sock
+    ensure
+      sock.close
+    end
+  end
+
```

Comment:
We could actually make this just `def self.open(*args, **nargs)` and then forward the arguments to `new`, so we only need one version of this method.